lib/AST/CASTBridging.cpp

```diff
@@ -365,51 +374,140 @@ void *BraceStmt_create(BridgedASTContext cContext, BridgedSourceLoc cLBLoc,
-      convertSourceLoc(cFuncLoc), declName, convertSourceLoc(cNameLoc), isAsync,
-      convertSourceLoc(cAsyncLoc), throws, convertSourceLoc(cThrowsLoc),
-      nullptr, paramList, (TypeRepr *)returnType, declContext);
+  auto *paramList = static_cast<ParameterList *>(opaqueParameterList);
+  auto declName = DeclName(context, convertIdentifier(cName), paramList);
+  auto asyncLoc = convertSourceLoc(cAsyncLoc);
+  auto throwsLoc = convertSourceLoc(cThrowsLoc);
+
+  auto *decl = FuncDecl::create(
+      context,
+      convertSourceLoc(cStaticLoc),
+      StaticSpellingKind::None,
+      convertSourceLoc(cFuncKeywordLoc),
+      declName,
+      convertSourceLoc(cNameLoc),
+      asyncLoc.isValid(),
+      asyncLoc,
+      throwsLoc.isValid(),
+      throwsLoc,
+      static_cast<GenericParamList *>(opaqueGenericParamList),
+      paramList,
+      static_cast<TypeRepr *>(opaqueReturnType),
+      convertDeclContext(cDeclContext));
+  decl->setTrailingWhereClause(
+      static_cast<TrailingWhereClause *>(opaqueGenericWhereClause));
+
+  return {bridgeDeclContext(decl), static_cast<Decl *>(decl)};
+}
+
+BridgedDeclContextAndDecl ConstructorDecl_create(
+    BridgedASTContext cContext,
+    BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cInitKeywordLoc,
+    BridgedSourceLoc cFailabilityMarkLoc,
+    bool isIUO,
+    void *_Nullable opaqueGenericParams,
+    void *opaqueParameterList,
+    BridgedSourceLoc cAsyncLoc,
+    BridgedSourceLoc cThrowsLoc,
+    void *_Nullable opaqueGenericWhereClause) {
+  assert((bool)cFailabilityMarkLoc.raw || !isIUO);
 
-  return {bridgeDeclContext(out), cast<FuncDecl>(out), cast<Decl>(out)};
-}
+  ASTContext &context = convertASTContext(cContext);
 
-void FuncDecl_setBody(void *fn, void *body) {
-  ((FuncDecl *)fn)->setBody((BraceStmt *)body, FuncDecl::BodyKind::Parsed);
+  auto *parameterList = static_cast<ParameterList *>(opaqueParameterList);
+  auto declName =
+      DeclName(context, DeclBaseName::createConstructor(), parameterList);
+  auto asyncLoc = convertSourceLoc(cAsyncLoc);
+  auto throwsLoc = convertSourceLoc(cThrowsLoc);
+  auto failabilityMarkLoc = convertSourceLoc(cFailabilityMarkLoc);
+
+  auto *decl = new (context) ConstructorDecl(
+      declName,
+      convertSourceLoc(cInitKeywordLoc),
+      failabilityMarkLoc.isValid(),
+      failabilityMarkLoc,
+      asyncLoc.isValid(),
+      asyncLoc,
+      throwsLoc.isValid(),
+      throwsLoc,
+      parameterList,
+      static_cast<GenericParamList *>(opaqueGenericParams),
+      convertDeclContext(cDeclContext));
+  decl->setTrailingWhereClause(
+      static_cast<TrailingWhereClause *>(opaqueGenericWhereClause));
+  decl->setImplicitlyUnwrappedOptional(isIUO);
+
+  return {bridgeDeclContext(decl), static_cast<Decl *>(decl)};
+}
+
+BridgedDeclContextAndDecl DestructorDecl_create(
+    BridgedASTContext cContext,
+    BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cDeinitKeywordLoc) {
```

Comment:
Ben persuaded me to reformat these using the config file that comes with the repo, so this is no longer relevant. For context, I originally formatted these using a custom config file of mine because [comment](https://github.com/apple/swift/pull/67111#discussion_r1264281821). Line breaks before `)` are off by default — you can turn them on.

Reply:
Ah sorry for not replying to that comment. I don't disagree with your points there, mostly I would just prefer consistency unless we have wider discussions on formatting. Going with what LLVM uses is just the simplest default right now 😅